nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+
+            try {
+                if (connectTimeout == 0) {
+                    connectTimeout = Long.MAX_VALUE;
+                }
+                binlogClient.connect(connectTimeout);
+
+            } catch (IOException | TimeoutException te) {
+                // Try the next host
+                connectedHost = null;
+                currentHost = (currentHost + 1) % numHosts;
+                connectionAttempts++;
+            }
+        }
+        if (!binlogClient.isConnected()) {
+            throw new IOException("Could not connect to any of the specified hosts");
+        }
+
+        if (createEnrichmentConnection) {
+            try {
+                jdbcConnection = getJdbcConnection(driverLocation, driverName, connectedHost, username, password, null);
+            } catch (InitializationException | SQLException e) {
+                throw new ProcessException(e);
+            }
+        }
+
+        doStop.set(false);
+    }
+
+
+    public void outputEvents(ProcessSession session, StateManager stateManager, ComponentLog log) throws IOException {
+        RawBinlogEvent rawBinlogEvent;
+
+        // Drain the queue
+        while ((rawBinlogEvent = queue.poll()) != null && !doStop.get()) {
+            Event event = rawBinlogEvent.getEvent();
+            EventHeaderV4 header = event.getHeader();
+            long timestamp = header.getTimestamp();
+            EventType eventType = header.getEventType();
+            // Advance the current binlog position. This way if no more events are received and the processor is stopped, it will resume at the event about to be processed.
+            // We always get ROTATE and FORMAT_DESCRIPTION messages no matter where we start (even from the end), and they won't have the correct "next position" value, so only
+            // advance the position if it is not that type of event. ROTATE events don't generate output CDC events and have the current binlog position in a special field, which
+            // is filled in during the ROTATE case
+            if (eventType != EventType.ROTATE && eventType != EventType.FORMAT_DESCRIPTION) {
+                currentBinlogPosition = header.getPosition();
+            }
+            log.debug("Got message event type: {} ", new Object[]{header.getEventType().toString()});
+            switch (eventType) {
+                case TABLE_MAP:
+                    // This is sent to inform which table is about to be changed by subsequent events
+                    TableMapEventData data = event.getData();
+
+                    // Should we skip this table? Yes if we've specified a DB or table name pattern and they don't match
+                    skipTable = (databaseNamePattern != null && !databaseNamePattern.matcher(data.getDatabase()).matches())
+                            || (tableNamePattern != null && !tableNamePattern.matcher(data.getTable()).matches());
+
+                    if (!skipTable) {
```

Comment:
Makes sense, I'll take a look, you're probably right, not clearing the currentTable had caused problems for me in other code paths too.